### PR TITLE
fix(core/select): clear button

### PR DESCRIPTION
**Pull request by goncalosard**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [ ] Unit tests (`yarn test`) were run locally and passed
- [x] Visual Regression Tests (`yarn visual-regression`) were run locally and passed
- [ ] Linting (`npm lint`) was run locally and passed

## Pull request type

Please check the type of change your PR introduces:

- [x] Bug fix
- [ ] Feature
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

GitHub Issue Number: #456 

## What is the new behavior?

- Clear button now works for Single select and Multiselect
- Adjusted the clear button size as designed

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by nuke-ellington — COMMENTED** [review]
Why does it not show up in VRT?

**Comment by goncalosard** [comment]
> Why does it not show up in VRT?

None of the existing examples had the property to use the clear button.
Added a vrt to test the clear button.

File: packages/core/src/components/select/select.tsx [hunk 1]
```diff
@@ -397,7 +397,7 @@ export class Select {
                   ref={(ref) => (this.inputRef = ref)}
                   onInput={() => this.filterItemsWithTypeahead()}
                 />
-                {this.allowClear &&
+                {this.allowClear === true &&
```
Comment (nuke-ellington) [inline packages/core/src/components/select/select.tsx:400]: Is this really neccessary?

File: packages/core/src/tests/select/select.e2e.ts
Comment (nuke-ellington) [inline packages/core/src/tests/select/select.e2e.ts:1]: please rename file/test case to allow-clear or something for more clarity

**Review by nuke-ellington — CHANGES_REQUESTED** [review]

File: packages/core/src/components/select/select.tsx (hunk 1 above)
Comment (goncalosard) [reply packages/core/src/components/select/select.tsx:400]: Without it, calling the prop on javascript would not work. "allow-clear" would not trigger the clear button to be displayed.

Comment (nuke-ellington) [reply packages/core/src/components/select/select.tsx:400]: Sorry I don't get it. Can you please explain in more detail why this won't work?

**Review by nuke-ellington — APPROVED** [review]
lgtm

**Pull request merged**